ddp-workspace/projects/ddp-sdk/src/lib/utility/RegExpHelper.ts

```diff
@@ -0,0 +1,50 @@
+export class RegExpHelper {
+
+    /* The method returns a RegExp:
+    *  if text includes symbols to substitute, they match with any of `substitutableSymbols` or `replaceValue`(a space by default),
+    *  (case/space insensitive)
+    *  e.g.
+    *  regExp = createSubstitutableSymbolsRegExp('asd / zxc', ['-', '/'], ' ');
+    *  regExp.test('asd zxc') = true;
+    *  regExp.test('asd / zxc') = true;
+    *  regExp.test('asd -    zxc') = true;
+    */
+    static createSubstitutableSymbolsRegExp(text: string, substitutableSymbols: string[] = [], replaceValue = ' '): RegExp {
+        const anyAmountOfSpacesRegExp = '\\s*';
+        if (substitutableSymbols.includes(text) || !substitutableSymbols.some(s => text.includes(s))) {
+            // return same text but space/case insensitive
+            return new RegExp(
+                RegExpHelper.replaceEmptySpaces(
+                    RegExpHelper.escapeRegExp(text),
+                    anyAmountOfSpacesRegExp
```

Comment:
why do we replace empty spaces with anyAmountOfSpacesRegExp?

Reply:
just for more handy search (not strict matching).
 Like: `the  (some mistyped spaces here)      brain` will match `the brain`


But, you are right. Changed a bit (peplaced any amount of spaces with one space, like `trim` method does), as to avoid such cases: `the  rwise` will match `otherwise`